modules/porous_flow/include/kernels/PorousFlowEffectiveStressCoupling.h

```diff
@@ -0,0 +1,54 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef POROUSFLOWEFFECTIVESTRESSCOUPLING_H
+#define POROUSFLOWEFFECTIVESTRESSCOUPLING_H
+
+#include "Kernel.h"
+
+#include "PorousFlowDictator.h"
+
+//Forward Declarations
+class PorousFlowEffectiveStressCoupling;
+
+template<>
+InputParameters validParams<PorousFlowEffectiveStressCoupling>();
+
+/**
+ * PorousFlowEffectiveStressCoupling computes -coefficient*effective_porepressure*grad_test[component]
+ */
+class PorousFlowEffectiveStressCoupling : public Kernel
+{
+public:
+
+  PorousFlowEffectiveStressCoupling(const InputParameters & parameters);
+
+ protected:
+
+  virtual Real computeQpResidual();
+
+  virtual Real computeQpJacobian();
+
```

Comment:
Take some of the empty lines and move them into the PorousFlowAdvectiveFlux header (that one looks pretty crammed) ;-)